src/language-js/print/statement.js

```diff
@@ -170,16 +170,18 @@ const isClassProperty = ({ type }) =>
   type === "ClassPrivateProperty" ||
   type === "ClassAccessorProperty" ||
   type === "AccessorProperty" ||
+  type === "TSAbstractPropertyDefinition" ||
   type === "TSAbstractAccessorProperty";
 /**
  * @returns {boolean}
  */
 function shouldPrintSemicolonAfterClassProperty(node, nextNode) {
-  const name = node.key && node.key.name;
+  const { name } = node.key;
   if (
     (name === "static" ||
       name === "get" ||
       name === "set" ||
+      // TODO: Remove this https://github.com/microsoft/TypeScript/issues/51707 is fixed
       name === "accessor") &&
     !node.value &&
     !node.typeAnnotation
```

Comment:
`printClassProperty` also prints `?` or `!` here https://github.com/prettier/prettier/blob/aa342098209be467e0fa2ffe8898243bcb7eaa0c/src/language-js/print/class.js#L229, not sure if we can omit semi here if those tokens exists, but we can investigate in feature.

We should also test private fields, since we use `.key.name` directly without check type.